crypto/dh/dh.h

```diff
@@ -356,7 +356,9 @@ int DH_KDF_X9_42(unsigned char *out, size_t outlen,
 
 /* KDF types */
 # define EVP_PKEY_DH_KDF_NONE                            1
+# ifndef OPENSSL_NO_CMS
```

Comment:
I [already commented on this in 1269](https://github.com/openssl/openssl/pull/1269#discussion_r184280601):

> Hmmm, we do exactly this in 1.1.0 and on... so for the sake of consistency, and also for those who use the presence of `EVP_PKEY_DH_KDF_X9_42` to determine if the functionality is available at all, this makes sense.

However, seeing that this is 1.0.2, it might be a bit on the late side to make this kind of change.